pyanaconda/isignal.py

```diff
@@ -0,0 +1,54 @@
+""" A signal/slot implementation
+
+File:    isignal.py
+Author:  Thiago Marcos P. Santos
+Author:  Christopher S. Case
+Author:  David H. Bronke
+Created: August 28, 2008
+Updated: December 12, 2011
+License: MIT
+
+"""
+import inspect
+from weakref import WeakKeyDictionary
+
+
+class Signal(object):
+    def __init__(self):
+        # The original implementation used WeakSet to store functions,
+        # but that causes lambdas without any other reference to be
+        # garbage collected. So we use a normal set to avoid that.
+        self._functions = set()
+        self._methods = WeakKeyDictionary()
+
+    def __call__(self, *args, **kargs):
```

Comment:
Maybe there could be `emit` method instead of `__call__`, because `signal.emit()` looks more clear to me than just `signal()`. For example Qt also uses emit: http://doc.qt.io/qt-4.8/signalsandslots.html